src/Tests/Microsoft.NET.Publish.Tests/GivenThatWeWantToPublishASingleFileApp.cs

```diff
@@ -651,5 +651,114 @@ public void It_errors_when_including_symbols_targeting_net5(bool selfContained)
                 .And
                 .HaveStdOutContaining(Strings.CannotIncludeSymbolsInSingleFile);
         }
+
+        [RequiresMSBuildVersionFact("16.8.0")]
+        public void It_errors_when_enabling_compression_targeting_net5()
+        {
+            var testProject = new TestProject()
+            {
+                Name = "SingleFileTest",
+                TargetFrameworks = "net5.0",
+                IsExe = true,
+            };
+            testProject.AdditionalProperties.Add("SelfContained", "true");
+
+            testProject.AdditionalProperties.Add("EnableCompressionInSingleFile", "true");
+
+            var testAsset = _testAssetsManager.CreateTestProject(testProject);
+            var publishCommand = new PublishCommand(testAsset);
+
+            publishCommand.Execute(PublishSingleFile, RuntimeIdentifier)
+                .Should()
+                .Fail()
+                .And
+                .HaveStdOutContaining(Strings.CompressionInSingleFileRequires60);
+        }
+
+        [RequiresMSBuildVersionFact("16.8.0")]
+        public void It_errors_when_enabling_compression_without_selfcontained()
+        {
+            var testProject = new TestProject()
+            {
+                Name = "SingleFileTest",
+                TargetFrameworks = "net6.0",
+                IsExe = true,
+            };
+
+            testProject.AdditionalProperties.Add("EnableCompressionInSingleFile", "true");
```

Comment:
You might have to explicitly set SelfContained to false since it defaults to true when there is a RuntimeIdentifier. I think you can also get rid of the exilicit SelfContained=true in the other tests.

Reply:
Ah, I see. I thought it is the opposite and it is not SelfContained unless explicitly specified. Thanks!

Reply:
@sbomer - do you know if there is an easy way to include a native lib dependency in these tests? I think the compression tests are failing because there is nothing in the bundle that we can compress. 

We do not compress managed assemblies yet. We had a bug that would cause all native libraries be included and tests were passing :-), but I think the fix for that got merged to SDK yesterday. Since all we have in the bundle is managed assemblies, we have nothing to compress thus the size of the app is unaffected by the compression setting.

Reply:
Not sure if there's a better way, but I found this test which adds a packagereference to sqlite and checks that native dependencies are included: https://github.com/dotnet/sdk/blob/main/src/Tests/Microsoft.NET.Build.Tests/GivenThatWeWantToCopyLocalDependencies.cs#L64

Reply:
This is not just tests - the only "truly supported" way to include native libraries (as in it works in all configs) is through nuget package reference. So the sqllite is as good as any for that purpose.

Reply:
I will try sqllite. Thanks.
It is point-in-time problem though. Once we compress managed assemblies, this will not be an issue, so if sqllite does not work I can also wait for that change.